### ISSUE TITLE
Re-initialize emulator when refitting in AL

### ISSUE THREAD

**Issue by radka-j** [alan-turing-institute/autoemulate#757]
This should follow [the suggestion ](https://github.com/alan-turing-institute/autoemulate/pull/752#issuecomment-3200001064)by @sgreenbury in #752

Similar approach is to be implemented in history matching workflow as part of #632 (see #748).

**Comment by sgreenbury** [comment]
For the AL tests, the learning rate was reduced in [c55b4e8](https://github.com/alan-turing-institute/autoemulate/pull/765/commits/c55b4e8e443533e2feb6136aeb9a6f5498d46406) to stability (not positive definite covariance matrices) when fitting the GP. It might also be good to explore how to set this hyperparameter in this issue for the AL workflow.
